lib/msf/core/session_event.rb

```diff
@@ -22,6 +22,7 @@ def on_session_close(session, reason='')
   # Called when the user interacts with a session.
   #
   def on_session_interact(session)
+    Rex::Ui::Text::Shell::HistoryManager.push_context(history_file: nil, name: session.type.to_sym)
```

Comment:
The `history_file` defaults to nil now so this can be reduced to

```ruby
    Rex::Ui::Text::Shell::HistoryManager.push_context(name: session.type.to_sym)
```